src/Common/TimerService.h

```diff
@@ -18,31 +24,47 @@ class TimerService final
     /// Safe to call from other threads.
     muduo::net::TimerId runAt(muduo::Timestamp time, muduo::net::TimerCallback cb)
     {
-        assert(event_loop);
+        std::scoped_lock lock(event_loop_mutex);
```

Comment:
I think we shall avoid using lock here since runAt is thread safe. Even event_loop itself is not thread safe, we require users to align `shutdown` and `runAt` which means shutdown is only invoked when system is going down and after that we shall never call `runAt`. If we do, there is logic error.

Reply:
This is why i ask if we can just mask TSAN for TimerService ? which means don't apply TSA to TimerService since we know what we are doing and for now, it seems not quite worth fixing it since we will decommission it.